pkg/informer/dynamic_discovery_shared_informer_factory.go

```diff
@@ -621,3 +658,142 @@ func (d *DynamicDiscoverySharedInformerFactory) Unsubscribe(id string) {
 
 	delete(d.subscribers, id)
 }
+
+func (d *DynamicDiscoverySharedInformerFactory) ServerGroups() (*metav1.APIGroupList, error) {
+	d.informersLock.RLock()
+	defer d.informersLock.RUnlock()
+
+	groups := make([]metav1.APIGroup, len(d.discoveryData))
+	failedGroups := make(map[schema.GroupVersion]error)
+
+	for i, apiResourceList := range d.discoveryData {
+		gv, err := schema.ParseGroupVersion(apiResourceList.GroupVersion)
+		if err != nil {
+			failedGroups[schema.GroupVersion{Group: apiResourceList.GroupVersion}] = err
+			continue
+		}
+		groups[i] = metav1.APIGroup{
+			Name: gv.Group,
+			Versions: []metav1.GroupVersionForDiscovery{
```

Comment:
it turns out it's required to include all possible versions. Otherwise owner references that point to a version other than the conflated / first served version, do not pass the coordinates check performed by the GC graph builder, as the node identity holds the wrong version, that is the conflated / first served one.

I've changed the index function in the dynamic shared informer factory to return all the versions, and completed the GC controller e2e tests to cover that case as well.